### DISCUSSION TITLE
ml-100k Data-set output embedding corresponding problem

### DISCUSSION THREAD

**Discussion by slightlyLLL** [RUCAIBox/RecBole#981] [Q&A] [answered]
Hello, developer, I have encountered a problem with the system developed by you. First of all, please forgive me for my academic inaccuracy. I want to get the embedding of users and items and save them, but I found that the saved vector is one more For example, there are 943 users in the ml-100k data set, but there are 944 vectors. How can I solve this?
![1](https://user-images.githubusercontent.com/44775326/135383553-a194d178-d1d2-4cd3-996a-c4b4676ed7ed.png)
![item](https://user-images.githubusercontent.com/44775326/135383586-273b1102-f78f-43ef-926b-3983aac83010.png)
![user](https://user-images.githubusercontent.com/44775326/135383590-13bf633e-3450-449d-9b71-fe80677623b9.png)

**Comment by Sherry-XLL** [comment, marked as answer]
@slightlyLLL Hello, RecBole will add `[PAD]` for all the token like fields. Thus after remapping ID, 0 will be reserved for `[PAD]`, which makes the result of `Dataset.item_num` and `Dataset.user_num` more than the actual number.

Note that for Knowledge-based models, we add one more relation called U-I Relation. It describes the history interactions which will be used in `recbole.data.dataset.kg_dataset.KnowledgeBasedDataset.ckg_graph()`. Thus the result of `KGDataset.relation_num` is two more than the actual number of relations.

Therefore, if you want to get the embeddings of users and items and save them, just skip the first embedding. Thanks for your attention to RecBole!
